### PR TITLE
[IMP] sale_product_configurator: redesign of the modal

### PR DESCRIPTION
**Pull request by anso-odoo**
This commit 
- Cleans and optimize the SCSS 
- Improves the layout and the spacing of the existing modal
- Improves the input's style (`.o_input`, better pills)
- Sets a fixed height to the modal so it doesn't grow/shrink when
fields are shown/hidden

| Before | After |
|--------|--------|
| Screenshot 2023-08-25 at 11 24 46 | Screenshot 2023-10-13 at 09 39 45 |

task-3289240
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by anso-odoo** [comment]
@stefanorigano Hello, can you have a look at this PR please ? Thanks !

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Hello @anso-odoo , thanks 👋
Sore preliminary remarks in the code

File: addons/sale_product_configurator/static/src/js/product/product.xml [hunk 1]
```diff
@@ -11,27 +11,34 @@
...
                 <PTAL t-props="ptal" productTmplId="this.props.product_tmpl_id"/>
             </t>
         </td>
-        <td t-if="!this.props.optional" class="text-center o_sale_product_configurator_qty">
-            <div class="input-group">
+        <td class="o_sale_product_configurator_price text-end" name="price">
+            <h5 class="text-nowrap" t-out="getFormattedPrice()"/>
+        </td>
+        <td class="o_sale_product_configurator_qty text-end">
+            <div t-if="!this.props.optional" class="input-group justify-content-end flex-nowrap border-0">
```
[the remark this answers is not in the document]
Comment (anso-odoo) [reply addons/sale_product_configurator/static/src/js/product/product.xml:33]: those are leftovers, will remove them :)

File: addons/sale_product_configurator/static/src/js/badge_extra_price/badge_extra_price.xml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <templates xml:space="preserve">
     <t t-name="product.badge_extra_price" owl="1">
-        <span class="badge rounded-pill text-bg-light border ms-1 ps-1">
-            <span t-out="this.props.price > 0 ? '+' : '-'" class="o_product_badge_extra_price"/>
-            <span t-out="getFormattedPrice()" class="text-muted fst-italic"/>
+        <span class="badge ms-1 ps-1 bg-100 rounded-pill">
```
[the remark this answers is not in the document]
Comment (anso-odoo) [reply addons/sale_product_configurator/static/src/js/badge_extra_price/badge_extra_price.xml:4]: I used a text-bg-info instead, I think it makes more sense than success as it's literally an "info" about the price 👍

File: addons/sale_product_configurator/static/src/js/product_list/product_list.xml [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
                     <th><span class="label"></span></th>
-                    <th class="text-center o_product_configurator_qty">
-                        <span class="label">Quantity</span>
-                    </th>
-                    <th class="text-center o_product_configurator_price">
+                    <th class="text-end o_product_configurator_price">
                         <span class="label">Price</span>
                     </th>
+                    <th class="o_product_configurator_qty text-end">
+                        <span class="label">Quantity</span>
```
[the remark this answers is not in the document]
Comment (anso-odoo) [reply addons/sale_product_configurator/static/src/js/product_list/product_list.xml:14]: It's leftovers from initial implementation, will remove that as it's useless 👍

File: addons/sale_product_configurator/static/src/js/badge_extra_price/badge_extra_price.xml [hunk 4]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <templates xml:space="preserve">
     <t t-name="product.badge_extra_price">
-        <span class="badge rounded-pill text-bg-light border ms-1 ps-1">
-            <span t-out="this.props.price > 0 ? '+' : '-'" class="o_product_badge_extra_price"/>
-            <span t-out="getFormattedPrice()" class="text-muted fst-italic"/>
+        <span class="badge ms-1 ps-1 text-bg-info rounded-pill">
+            <span t-out="this.props.price > 0 ? '+' : '-'" class="o_product_badge_extra_price me-1"/>
```
Comment (chevalierv) [inline addons/sale_product_configurator/static/src/js/badge_extra_price/badge_extra_price.xml:5]: Why do you keep the class if you remove the SCSS associated with it?
```suggestion
            <span t-out="this.props.price > 0 ? '+' : '-'" class="me-1"/>
```

File: addons/sale_product_configurator/static/src/js/product/product.xml [hunk 5]
```diff
@@ -1,70 +1,73 @@
...
                 t-else=""
+                class="w-100"
                 t-att-src="'/web/image/product.template/'+this.props.product_tmpl_id+'/image_128'"
                 alt="Product Image"/>
         </td>
-        <td t-att-colspan="this.props.optional ? 2:false">
-            <div class="o_sale_product_configurator_name">
-                <strong t-out="this.props.display_name"/>
+        <td class="p-3" t-att-colspan="this.props.optional ? 2:false">
+            <div class="o_sale_product_configurator_name mb-3 text-break">
```
Comment (chevalierv) [inline addons/sale_product_configurator/static/src/js/product/product.xml:17]: same question 
```suggestion
            <div class="mb-3 text-break">
```

**Comment by anso-odoo** [comment]
@stefanorigano Hello, can you have a second look at this when you have time pls ? Ty !

File: addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.scss [hunk 6]
```diff
@@ -66,27 +47,11 @@
...
-        background-color: map-get($grays, '200');
-    }
-
-    input {
-        -moz-appearance: none;
-        -webkit-appearance: none;
-        appearance: none;
-        opacity: 0;
-    }
+.o_sale_product_configurator_ptav_pills.active label{
```
Comment (stefanorigano) [inline addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.scss:50]:
```suggestion
.o_sale_product_configurator_ptav_pills.active label {
```

File: addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.scss [hunk 7]
```diff
@@ -66,27 +47,11 @@
...
+    $-btn-secondary-design: map-get($o-btns-bs-override, "secondary");
```
Comment (stefanorigano) [inline addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.scss:51]: Missing new line after variables definition:
```suggestion
    $-btn-secondary-design: map-get($o-btns-bs-override, "secondary");
    
```

**Review by stefanorigano — CHANGES_REQUESTED** [review]
@anso-odoo , almost there, courage 💪

File: addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.xml [hunk 8]
```diff
@@ -2,26 +2,33 @@
...
+            <div class="d-flex flex-column flex-lg-row gap-2 mb-2">
+                <label
+                    t-if="this.props.attribute_values.length === 1 &amp;&amp; isSelectedPTAVCustom() || this.props.attribute_values.length &gt; 1"
+                    t-out="this.props.attribute.name"
+                    t-attf-class="fw-bold text-break #{this.props.attribute_values.length === 1 &amp;&amp; hasPTAVCustom() ? '' : 'w-lg-25'}"/>
+                <t t-if="this.props.attribute_values.length > 1" t-call="{{getPTAVTemplate()}}"/>
+            </div>
+            <div class="w-75 mb-4 ms-auto ps-2" t-if="hasPTAVCustom()">
+                <input
+                    t-attf-class="o_input #{!isSelectedPTAVCustom() ? 'opacity-50' : ''}"
```
[the remark this answers is not in the document]
Comment (anso-odoo) [reply addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.xml:19]: The input shouldn't disappear but should appear disabled
Not sure how we manage the disabled input states in the backend. It does nothing with only `o_input` and a `disabled` attribute so I tried to use a `form-control` + `disabled` but then it adds a solid background.

[the remark this answers is not in the document]
Comment (stefanorigano) [reply addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.xml:19]: Normally we hide fields that are not editable.

[the remark this answers is not in the document]
Comment (anso-odoo) [reply addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.xml]: @dalaOdoo Hi, please can you check this comment 
Considering hiding the disabled inputs is the default behaviour in the back-end, i'm resetting to the previous behaviour. Also added a fixed height to the modal if there is at least an optional product so it's doesn't grow/shrink when we show/hide fields. Can you have a look on runbot please ? 
Thanks

[the remark this answers is not in the document]
Comment (dalaOdoo) [reply addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.xml]: @anso-odoo I'll test it again when it is built.
@stefanorigano My concern is the flickering when selecting "custom" as we will have to display the input. For me the input is part of "custom" choice and so it is not a problem displaying it even if custom is not selected

[the remark this answers is not in the document]
Comment (dalaOdoo) [reply addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.xml]: @anso-odoo still not fan but I'll not block for this

[the remark this answers is not in the document]
Comment (dalaOdoo) [reply addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.xml]: Conversely, if it is the only thing that is blocking this PR I thing we should just keep the version that is working (visible and disabled ?) and  move on. 
This should not be considered as blocking, don't waste time now on nitpicking comment.

CC @anso-odoo @stefanorigano

[the remark this answers is not in the document]
Comment (anso-odoo) [reply addons/sale_product_configurator/static/src/js/product_template_attribute_line/product_template_attribute_line.xml]: After discussion with @dalaOdoo, we will continue with the default system of hiding disabled fields
I'm marking this PR as ready for review @chevalierv

**Comment by Demesmaeker** [comment]
> @stefanorigano The tours are currently broken but @chevalierv is having a look to fix them 👍

Hello, tours should be fixed now ^^
Have a nice day!

cc: @chevalierv

**Comment by Feyensv** [comment]
Still red :eyes:

@robodoo delegate=chevalierv

**Comment by Demesmaeker** [comment]
Yes, just noticed it too xD I forgot event with the product configurator, incoming =]

**Comment by anso-odoo** [comment]
Hey @chevalierv, looks like it's all 🟢 👍

**Comment by chevalierv** [comment]
@robodoo r+

**Comment by chevalierv** [comment]
@robodoo r-

**Comment by chevalierv** [comment]
@robodoo r+

**Comment by vava-odoo** [comment]
Hello @anso-odoo 
Could you backport the commit to 17.0? This PR was merged after the freeze.
Thanks!

**Comment by AntoineVDV** [comment]
Backported with https://github.com/odoo/odoo/pull/140096

**Pull request closed without merging**